### PR TITLE
feat: added gosec G104

### PR DESCRIPTION
**Pull request by amarps**
## Description
Related to #3545, added gosec G104

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### CODE REVIEW

**Review by janisz — CHANGES_REQUESTED** [review]
CI is failing

**Comment by amarps** [comment]
i can't run `golangci-lint run` locally on this project. im using go version go1.19.2 windows/amd64

```level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\pkg\\fsutils\\utils.go:29:42: Statfs_t not declared by package syscall"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\pkg\\fsutils\\utils.go:30:19: Statfs_t not declared by package syscall"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\pkg\\fsutils\\utils.go:31:20: Statfs not declared by package syscall"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\central\\globaldb\\v2backuprestore\\backup\\generators\\dbs\\bolt.go:9:2: could not import github.com/stackrox/rox/pkg/odirect (-: build constraints exclude all Go files in C:\\code\\opensources\\stackrox\\pkg\\odirect)"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\central\\globaldb\\export\\restore.go:17:2: could not import github.com/stackrox/rox/pkg/odirect (-: build constraints exclude all Go files in C:\\code\\opensources\\stackrox\\pkg\\odirect)"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\central\\globaldb\\v2backuprestore\\formats\\roxdbv1\\bolt.go:10:2: could not import github.com/stackrox/rox/pkg/odirect (-: build constraints exclude all Go files in C:\\code\\opensources\\stackrox\\pkg\\odirect)"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\central\\main.go:417:21: Kill not declared by package syscall"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\compliance\\collection\\file\\file.go:220:28: Stat_t not declared by package syscall"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\compliance\\collection\\file\\file.go:221:28: Stat_t not declared by package syscall"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\migrator\\compact\\compact.go:117:20: undeclared name: mmapFlags"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\sensor\\admission-control\\main.go:49:27: SIGTERM not declared by package unix"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\sensor\\admission-control\\main.go:49:41: SIGINT not declared by package unix"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\sensor\\admission-control\\main.go:122:19: SIGTERM not declared by package unix"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\sensor\\kubernetes\\main.go:38:41: SIGTERM not declared by package unix"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\central\\cluster\\datastore\\datastore_impl.go:514:14: invalid operation: cannot compare ds.cm != nil (operator != not defined on untyped nil)"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\central\\compliance\\datastore\\test\\datastore_test.go:40:12: undeclared name: DataStore"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\central\\compliance\\datastore\\test\\datastore_test.go:153:12: undeclared name: DataStore"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\pkg\\dackbox\\tests\\cached_search_test.go:103:15: undeclared name: NewCachedSearcher"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\pkg\\dackbox\\tests\\cached_search_test.go:120:15: undeclared name: NewCachedSearcher"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\pkg\\dackbox\\tests\\cached_search_test.go:137:15: undeclared name: NewCachedSearcher"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\pkg\\rocksdb\\rocksdb.go:22:8: could not import C (cgo preprocessing failed)"
level=error msg="[linters_context] typechecking error: C:\\code\\opensources\\stackrox\\pkg\\rocksdb\\close_test.go:36:17: db.Get undefined (type *RocksDB has no field or method Get)"
level=error msg="[linters_context] typechecking error: C:\\Program Files\\Go\\src\\runtime\\cgo\\cgo.go:33:8: could not import C (cgo preprocessing failed)"
level=warning msg="[runner] Can't run linter goanalysis_metalinter: buildir: failed to load package gorocksdb: could not load export data: no export data for \"github.com/tecbot/gorocksdb\""
level=error msg="Running error: 1 error occurred:\n\t* can't run linter goanalysis_metalinter: buildir: failed to load package gorocksdb: could not load export data: no export data for \"github.com/tecbot/gorocksdb\"\n\n"```

File: central/activecomponent/updater/updater_impl.go [hunk 1]
```diff
@@ -144,6 +144,7 @@ func (u *updaterImpl) updateForDeployment(ctx context.Context, deploymentID stri
 
 		result, ok := u.executableCache.Get(update.ImageID)
 		if !ok {
+			//#nosec G104
```
Comment (janisz) [inline central/activecomponent/updater/updater_impl.go:147]: I think we should have a follow up PR to replace utils.Should with function that does not return error. Adding `#nosec` directives makes no sense

Comment (0x656b694d) [reply central/activecomponent/updater/updater_impl.go:147]: There are tens of places where the return value is used. So there are two options: either implement two functions, or change the occurrences in this PR to `_ = utils.Should(...`.

**Review by janisz — CHANGES_REQUESTED** [review]
CI is failing

**Comment by 0x656b694d** [comment]
Please rebase to master as https://github.com/stackrox/stackrox/pull/3830 has been merged.

**Comment by janisz** [comment]
Fixed by https://github.com/stackrox/stackrox/pull/3936

**Pull request closed without merging**